### DISCUSSION TITLE
Ideas on Conversion of Date to BsonTypeDate

### DISCUSSION THREAD

**Discussion by ysnarafat** [foyzulkarim/mern-video-streaming#58] [Ideas]
While working on issue [#54](https://github.com/foyzulkarim/mern-video-streaming/issues/54), I encountered a problem when attempting to insert a value into the `recordingDate` field via a postman request. To verify the deletion functionality, I tried inserting a value and then deleting it using its ID.

The validation error occurred whenever I passed `recordingDate` as a normal date from the request body. To address this issue, I manually parsed the date string as a date object and passed it to the insert method in the collection (generic repository).

Now, the question is how to elegantly convert a human-readable date to `bsonType: date`. This conversion is essential to handle the `recordingDate` field correctly.

**Comment by foyzulkarim** [comment]
Will look into this. Thanks.
